File: codigo-fonte/scala/somar-numeros/SomarNumeros.scala
object SomarNumeros extends App {
  val tempoInicial: Double = System.nanoTime

  def somar(inicio: Long, total: Long, resultado: Long = 0): Long =
    if (inicio >= total) resultado else somar(inicio + 1, total, inicio + resultado)

  val resultado = somar(1, 1000000)
  val tempoFinal: Double = System.nanoTime

  println(s"Resultado: $resultado")
  println(s"Tempo (ms): ${((tempoFinal - tempoInicial) / 1000000)}")
}

File: codigo-fonte/java/somar-numeros/SomarNumeros.java
public class SomarNumeros {

	public static void main(String[] args) {
		double tempoInicial = System.nanoTime();

		long somatoria = 0;
		for (long i = 1; i < 1000000; i++) {
			somatoria += i;
		}

		double tempoFinal = System.nanoTime();

		System.out.println("Resultado: " + somatoria);
		System.out.println("Tempo (ms): " + ((tempoFinal - tempoInicial) / 1000000));
	}

}